File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution 
{
    public:
    int minCost(string colors, vector<int>& nt) 
    {
        colors += (colors[colors.size()-1]=='a' ? 'b' : 'a');
        nt.push_back(1);
        int l = nt.size();
        int s = nt[0], maxn = nt[0], res = 0;
        for(int i=1; i<l; i++)
        {
            if(colors[i] == colors[i-1])
            {
                s += nt[i];
                maxn = max(maxn, nt[i]);
            }
            else
            {
                s -= maxn;
                res += s;
                s = nt[i];
                maxn = nt[i];  
            }
        }
        return res;
    }
};

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) 
    {
        colors += " ";
        int nt[] = new int[neededTime.length+1];
        for(int i=0; i<=neededTime.length; i++)
        {
            if(i==neededTime.length)
                nt[i] = 1;
            else
                nt[i] = neededTime[i];
        }
        int l=nt.length, s=nt[0], maxn=nt[0], res=0;
        for(int i=1; i<l; i++)
        {
            if(colors.charAt(i) == colors.charAt(i-1))
            {
                s += nt[i];
                maxn = Math.max(maxn, nt[i]);
            }
            else
            {
                res += s-maxn;
                s = maxn = nt[i];
            }
        }
        return res;    
    }
}